BossExpertise.cs

```diff
@@ -54,48 +64,83 @@ public override void HandlePacket(BinaryReader reader, int whoAmI)
 			}
 		}
 		
-		void SyncExpertMode(bool expert, int ignoreClient = -1)
-		{
-			HookExpertMode(expert);
-			if(Main.netMode != NetmodeID.SinglePlayer)
+		void SyncDifficultyMode(int difficulty, int ignoreClient = -1)
+		{ 
+			HookDifficultyMode(difficulty);
+			if (Main.netMode != NetmodeID.SinglePlayer)
 			{
 				var msg = GetPacket();
-				msg.Write((byte)ExpertMessageType.SyncExpert);
-				msg.Write(expert);
+				msg.Write((byte)ExpertMessageType.SyncDifficulty);
+				msg.Write(difficulty);
 				msg.Send(ignoreClient: ignoreClient);
 			}
 		}
 		
-		public static void HookExpertMode(bool? givenValue)
+		public static void HookDifficultyMode(int difficulty)
         {
-			FieldInfo expert = typeof(Main)
+			if (difficulty == 0 || difficulty > 2)
+            {
+				FieldInfo expert = typeof(Main)
+				.GetField("_overrideForExpertMode", BindingFlags.Static | BindingFlags.NonPublic);
+				expert.SetValue(null, false);
+				FieldInfo master = typeof(Main)
+				.GetField("_overrideForMasterMode", BindingFlags.Static | BindingFlags.NonPublic);
+				master.SetValue(null, false);
+			}
+			else if (difficulty == 1)
+			{
+				FieldInfo expert = typeof(Main)
 				.GetField("_overrideForExpertMode", BindingFlags.Static | BindingFlags.NonPublic);
-				expert.SetValue(null, givenValue);
+				expert.SetValue(null, true);
```

Comment:
Maybe better to set master mode to false, just in case

Reply:
Done.